src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsysreflect/QSysReflectCommunicator.java

```diff
@@ -587,6 +701,38 @@ private void retrieveDevices() {
 						.get(QSysReflectConstant.MESSAGE).asText());
 			}
 			aggregatedDeviceList = new ArrayList<>(aggregatedDeviceProcessor.extractDevices(devices));
+			for (AggregatedDevice aggregatedDevice : aggregatedDeviceList) {
+				Map<String, String> stats = aggregatedDevice.getProperties();
+				stats.put(QSysReflectConstant.DEVICE_TYPE, QSysReflectConstant.CORE);
+				aggregatedDevice.setProperties(stats);
+			}
+		} catch (Exception e) {
+			String errorMessage = String.format("Aggregated Device Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage());
+			deviceErrorMessagesList.add(errorMessage);
+			logger.error(errorMessage);
```

Comment:
can you please add an original error here as well? Just not to swallow and lose it